sky/task.py

```diff
@@ -485,13 +494,42 @@ def update_storage_mounts(self, storage_mounts: Dict[str,
 
     def add_storage_mounts(self) -> None:
```

Comment:
See the other comment about renaming this. @romilbhardwaj 

(This is actually uploading things. Its name should reflect that to warn callers.)

Reply:
Yes, let's change the name for now and maybe add a todo stating that this should be lazy